### PR TITLE
ESM: revisit need for `_VSCODE_NODE_MODULES` (fix #226453)

### LINKED ISSUE #226453: ESM: revisit need for `_VSCODE_NODE_MODULES`
A reminder that we now can maybe cleanup this:

https://github.com/microsoft/vscode/blob/091668b640caf96543ae8e2604f020aceee2de2d/src/typings/vscode-globals-modules.d.ts#L10-L28

Renderers are node-free and shared process is node.js enabled utility process (without workers).

We still seem to be using the global in some contexts though, but that seems to be replaceable with a `require`?

### PR DESCRIPTION
**Pull request by bpasero**

### CODE REVIEW

File: src/vs/server/node/remoteExtensionHostAgentServer.ts [hunk 1]
```diff
@@ -768,7 +768,7 @@ export async function createServer(address: string | net.AddressInfo | null, arg
 		const hasVSDA = fs.existsSync(join(FileAccess.asFileUri('').fsPath, '../node_modules/vsda'));
 		if (hasVSDA) {
 			try {
-				return <typeof vsda>globalThis._VSCODE_NODE_MODULES['vsda'];
+				return require('vsda');
```
Comment (jrieken) [inline src/vs/server/node/remoteExtensionHostAgentServer.ts:771]: Would we need `const require = createRequire(import.meta.url);` in this file?

File: src/vs/amdX.ts [hunk 2]
```diff
@@ -185,9 +185,9 @@ class AMDModuleImporter {
 
 	private async _nodeJSLoadScript(scriptSrc: string): Promise<DefineCall | undefined> {
 		try {
-			const fs = (globalThis as any)._VSCODE_NODE_MODULES['fs'];
-			const vm = (globalThis as any)._VSCODE_NODE_MODULES['vm'];
-			const module = (globalThis as any)._VSCODE_NODE_MODULES['module'];
+			const fs = (await import('fs')).default;
+			const vm = (await import('vm')).default;
+			const module = (await import('module')).default;
```
Comment (jrieken) [inline src/vs/amdX.ts:190]: needs some any-casts to make the monaco compile check happy.

Comment (bpasero) [reply src/vs/amdX.ts:190]: yes

File: src/vs/server/node/remoteExtensionHostAgentServer.ts (hunk 1 above)
Comment (bpasero) [reply src/vs/server/node/remoteExtensionHostAgentServer.ts:771]: indeed

**Review by chrmarti — APPROVED** [review]

**Pull request merged**